File: app/front/src/components/main/dto/add_xml/updateWkTblAddByXmlTableListCapsuleDto.ts
import type { WkTblMasterAllByXmlEntityInterface } from "../../entity/wkTblMasterAllByXmlEntity";
import { type FrameworkCapsuleDtoInterface ,FrameworkCapsuleDto } from "seijishikin-jp-normalize_common-tool";

interface UpdateWkTblAddByXmlTableListCapsuleDtoInterface extends FrameworkCapsuleDtoInterface {

    /** 編集対象リスト */
    listWkTblByXml: WkTblMasterAllByXmlEntityInterface[];

}

class UpdateWkTblAddByXmlTableListCapsuleDto extends FrameworkCapsuleDto
    implements UpdateWkTblAddByXmlTableListCapsuleDtoInterface {

    /** 編集対象リスト */
    listWkTblByXml: WkTblMasterAllByXmlEntityInterface[];

    constructor() {
        super();
        this.listWkTblByXml = [];
    }
}

export { type UpdateWkTblAddByXmlTableListCapsuleDtoInterface, UpdateWkTblAddByXmlTableListCapsuleDto }

File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/add_xml/UpdateWkTblAddByXmlTableListCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.add_xml;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import net.seijishikin.jp.normalize.common_tool.dto.FrameworkCapsuleDto;
import net.seijishikin.jp.normalize.manage.kanrensha.entity.WkTblMasterAllByXmlEntity;


/**
 * XML一括登録最小マスタ編集リストワークテーブル更新CapsuleDto
 */
public class UpdateWkTblAddByXmlTableListCapsuleDto extends FrameworkCapsuleDto // NOPMD DataClass
        implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 編集対象リスト */
    private List<WkTblMasterAllByXmlEntity> listWkTblByXml = new ArrayList<>();

    /**
     * 編集対象リストを取得する
     *
     * @return 編集対象リスト
     */
    public List<WkTblMasterAllByXmlEntity> getListWkTblByXml() {
        return listWkTblByXml;
    }

    /**
     * 編集対象リストを設定する
     *
     * @param listWkTblByXml 編集対象リスト
     */
    public void setListWkTblByXml(final List<WkTblMasterAllByXmlEntity> listWkTblByXml) {
        this.listWkTblByXml = listWkTblByXml;
    }

}
